src/bitcoin/block.h

```diff
@@ -77,26 +173,33 @@ class CBlock : public CBlockHeader {
     ADD_SERIALIZE_METHODS
 
     template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream &s, Operation ser_action) {
+        inline void SerializationOp(Stream &s, Operation ser_action) {
         READWRITEAS(CBlockHeader, *this);
+        READWRITE(vMetadata);
         READWRITE(vtx);
     }
 
     void SetNull() {
         CBlockHeader::SetNull();
         vtx.clear();
+        vMetadata.clear();
         fChecked = false;
     }
 
     CBlockHeader GetBlockHeader() const {
-        CBlockHeader block;
-        block.nVersion = nVersion;
-        block.hashPrevBlock = hashPrevBlock;
-        block.hashMerkleRoot = hashMerkleRoot;
-        block.nTime = nTime;
-        block.nBits = nBits;
-        block.nNonce = nNonce;
-        return block;
+        CBlockHeader header;
```

Comment:
This is bizarro to me. Original bitcoin code here is bizarro too. Can just do `return *this` here and it would work and not be error-prone.  I don't see a reason for this since this is setting up all the fields as a copy-construction would.